Reaction expand to comment list's top rather than to the bottom.

desyncr opened the pull request:
Add logic to show comment form or reactions depending on execution path
- if the user clicks 'reactions' link the reactions view is displayed
  - the reactions view displays a button to show the comment form
  - the screen is scrolled to the top of the reactions view
- if the user clicks the 'pen' icon the reactions view is displayed
  - the reactions view displays the comment form and is focused
  - the screen is scrolled to the bottom of the reactions view

Added logic to displayed the 'comment' button after successfully commenting
- when the user comments on a post
  - the comment form is hidden
  - the 'comment' button is shown again

Fixed #5757 btw
edit: Updated description to reflect logic changes

Commit:
Reaction expand to comment list's top rather than to the bottom.

Added method 'scrollToCommentsList' to scroll to the top of the
comment list ('.bottom_bar'), instead of scrolling to the last comment.

Flaburgan commented:
Hey, thanks for working on this.
This is not exactly the behavior we want. On the mobile view, there are four links in the bottom of a post: "N reaction", and then the reshare, pen and like icon. When clicking on the "reaction" link, the user wants to display the reaction. That's why the reactions have to appear and the link should be an anchor to the first reaction. When clicking on the "pen" icon, the user wants to comment the post. That's why the reaction should appear but the focus should be directly at the bottom, on the textarea allowing to comment the post.

desyncr commented:
So it needs to scroll to the very bottom  And focus on the input on the pen action? Anything else?

I'll look into it tonight.

Commit:
Add logic to show comment form or reactions depending on execution path

- if the user clicks 'reactions' link the reactions view is displayed
    - the reactions view displays a button to show the comment form
    - the screen is scrolled to the top of the reactions view
- if the user clicks the 'pen' icon the reactions view is displayed
    - the reactions view displays the comment form and is focused
    - the screen is scrolled to the bottom of the reactions view

Added logic to displayed the 'comment' button after successfuly
commenting
- when the user comments on a post
    - the comment form is hidden
    - the 'comment' button is shown again

Commit:
Fix coding standard errors

Commit:
Fix coding standard errors

desyncr commented:
@Flaburgan Is the logic implemented following the expected behavior?

Commit:
Update hash syntax and feature tests

Commit:
Fix list item style on FF

Flaburgan commented on app/assets/javascripts/mobile/mobile.js:
```diff
@@ -125,57 +125,83 @@ $(document).ready(function(){
-        link.removeClass('active');
+    var initializeReactionsView = function() {
+      // Update 'reactions' link and 'pen' button to reflect the current state
+      activeElements.removeClass('inactive').addClass('active');
```
I'm wondering why we need two class here, only one and using toggle should be enough. This would imply to also refactor the css though so if you don't want to touch it, that's fine.

Flaburgan commented on app/assets/javascripts/mobile/mobile.js:
```diff
@@ -125,57 +125,83 @@ $(document).ready(function(){
+      activeElements.removeClass('active').addClass('inactive');
 
-    } else if( existingCommentsContainer.length > 0) {
+      $('html,body').scrollTop(parent.offset().top - parent.closest(".stream_element").height() - 8);
```
Why can't an anchor be used here? This looks like a hack to me.

desyncr replied:
No reason. It was the original approach.

RichKavanagh commented:
@desyncr @houndci @Flaburgan Thanks for working on this one guys!

RichKavanagh commented:
Does this mean the PR was rejected?

denschub commented:
The original author decided to delete his branch and close this pull request. There is nothing we could possibly do.

RichKavanagh commented:
Ah right, I see. So what should I do now, open a new issue as a feature request or as a bug?

theworldbright commented:
I've implemented a fix for my own pod here that includes this issue: https://github.com/diaspora/diaspora/pull/6066. It works for me on the Chrome mobile simulator and the Nexus 5 but no maintainer has done a functionality check yet so be wary.

RichKavanagh commented:
If your fix that was merged works then you'd better claim the bounty mate :)

theworldbright commented:
@RichKavanagh thanks
